lld/lld-c/LLDAsLibraryC.cpp

```diff
@@ -896,3 +898,69 @@ LLVMBool LLVMLinkEVM(LLVMMemoryBufferRef inBuffers[],
 
   return false;
 }
+
+/// Returns immutables and their offsets of the ELF object
+/// file passed in \p inBuffer.
+uint64_t LLVMGetImmutablesEVM(LLVMMemoryBufferRef inBuffer,
+                              char ***immutableIDs,
+                              uint64_t **immutableOffsets) {
+  std::unique_ptr<Binary> inBinary =
+      cantFail(createBinary(unwrap(inBuffer)->getMemBufferRef()));
+  const auto *objFile = static_cast<const ObjectFile *>(inBinary.get());
+
+  // Maps immutable IDs to their references in the object code.
+  StringMap<SmallVector<uint64_t>> immutablesMap;
+  for (const SymbolRef &sym : objFile->symbols()) {
+    section_iterator symSec = cantFail(sym.getSection());
+    if (symSec == objFile->section_end())
+      continue;
+
+    StringRef symName = cantFail(sym.getName());
+    if (EVM::isLoadImmutableSymbolName(symName)) {
+      std::string Id = EVM::getImmutableId(symName);
+      uint64_t symOffset = cantFail(sym.getValue());
+      // The symbol points to the beginning of a PUSH32 instruction.
+      // We have to add 1 (opcode size) to get offset to the PUSH32
+      // instruction operand.
+      immutablesMap[Id].push_back(symOffset + 1);
+    }
+  }
+
+  if (immutablesMap.empty()) {
+    *immutableIDs = nullptr;
+    *immutableOffsets = nullptr;
+    return 0;
+  }
+
+  uint64_t numOfImmutables = 0;
```

Comment:
Nit: a loop would look more natural here, but if you preffer an algorithm why not accumulate?